File: src/main/scala/com/lcy/scala/spark/rdd/secondarySort/SecondarySortApp.scala
package com.lcy.scala.spark.rdd.secondarySort

import org.apache.spark.{SparkConf, SparkContext}

//二次排序
object SecondarySortApp {

  def main(args: Array[String]): Unit = {
    val conf = new SparkConf().setAppName("SecondSort").setMaster("local")

    val sc = new SparkContext(conf)

    sc.textFile("file:\\D:\\sparkData\\demo3.txt")
      .map(line =>{
        val str = line.split(" ")
        (new MySecondSort(str(1).trim.toInt, str(0).trim), line)
      })
      .sortByKey()
      .map(pair => pair._2)
      .foreach(println)
  }

}


/**
  * 数据
  * a 2
  * a 2
  * a 3
  * b 3
  * b 4
  * b 4
  */

File: src/main/java/com/lcy/java/spark/rdd/secondarySort/SecondarySortApp.java
package com.lcy.java.spark.rdd.secondarySort;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaSparkContext;
import scala.Tuple2;
import scala.math.Ordered;

import java.io.Serializable;
import java.util.Objects;

public class SecondarySortApp {
    
    public static void main(String[] args) {
        
        SparkConf conf = new SparkConf().setAppName("SecondarySortApp").setMaster("local");
        JavaSparkContext sc = new JavaSparkContext(conf);
        
        sc.textFile("file:\\D:\\sparkData\\demo3.txt")
                .mapToPair(line -> {
                    String[] lineSplits = line.split(" ");
                    return new Tuple2<>(new SecondSort(Integer.valueOf(lineSplits[0]),
                            Integer.valueOf(lineSplits[1])), line);
                })
                .sortByKey()
                .foreach(res -> System.out.println(res._2));
        
        sc.close();
        
    }
    
}

/**
 * 数据
 * 12 23
 * 12 34
 * 23 34
 * 23 45
 * 45 23
 * 56 23
 */
